### ISSUE TITLE
[RyuJIT/armel] HFA tests fail

### ISSUE THREAD

**Issue by wateret** [dotnet/coreclr#13312]
With #13284, most regressions caused by #13023 were fixed. But still there is HFA test regression.

All of these run successfully, but the fails with Unexpected Results. (wrong result value)

- JIT/jit64/hfa/main/testB/hfa_sf2B_d/hfa_sf2B_d.exe
- JIT/jit64/hfa/main/testB/hfa_sf0B_r/hfa_sf0B_r.exe
- JIT/jit64/hfa/main/testB/hfa_sf2B_r/hfa_sf2B_r.exe
- JIT/jit64/hfa/main/testG/hfa_sd0G_r/hfa_sd0G_r.exe
- JIT/jit64/hfa/main/testG/hfa_sd1G_r/hfa_sd1G_r.exe
- JIT/jit64/hfa/main/testG/hfa_sd2G_r/hfa_sd2G_r.exe
- JIT/jit64/hfa/main/testG/hfa_sf2G_r/hfa_sf2G_r.exe
- JIT/jit64/hfa/main/testA/hfa_sd0A_r/hfa_sd0A_r.exe
- JIT/jit64/hfa/main/testA/hfa_sd1A_r/hfa_sd1A_r.exe
- JIT/jit64/hfa/main/testA/hfa_sd2A_d/hfa_sd2A_d.exe
- JIT/jit64/hfa/main/testA/hfa_sd2A_r/hfa_sd2A_r.exe
- JIT/jit64/hfa/main/testA/hfa_sf0A_r/hfa_sf0A_r.exe
- JIT/jit64/hfa/main/testA/hfa_sf1A_r/hfa_sf1A_r.exe
- JIT/jit64/hfa/main/testA/hfa_sf2A_d/hfa_sf2A_d.exe
- JIT/jit64/hfa/main/testA/hfa_sf2A_r/hfa_sf2A_r.exe
- JIT/jit64/hfa/main/testB/hfa_sd0B_d/hfa_sd0B_d.exe
- JIT/jit64/hfa/main/testB/hfa_sd0B_r/hfa_sd0B_r.exe
- JIT/jit64/hfa/main/testB/hfa_sd2B_d/hfa_sd2B_d.exe
- JIT/jit64/hfa/main/testB/hfa_sd2B_r/hfa_sd2B_r.exe
- JIT/jit64/hfa/main/testB/hfa_sf0B_d/hfa_sf0B_d.exe

**Comment by wateret** [comment]
Result of `JIT/jit64/hfa/main/testA/hfa_sd0A_r/hfa_sd0A_r.exe`

```
[Sum11 HFA 01]  PASSED
[Sum11 HFA 02]  FAILED => expected = 3, actual = 80
[Sum11 HFA 03]  FAILED => expected = 6, actual = 83
[Sum11 HFA 05]  PASSED
[Sum11 HFA 08]  PASSED
[Sum11 HFA 11]  PASSED
[Sum11 HFA 19]  PASSED
[Sum19 HFA 01]  PASSED
[Sum19 HFA 02]  PASSED
[Sum19 HFA 03]  PASSED
[Sum19 HFA 05]  PASSED
[Sum19 HFA 08]  PASSED
[Sum19 HFA 11]  PASSED
[Sum19 HFA 19]  PASSED
```

**Comment by wateret** [comment]
This is the first part of JitDump gets different. (TC: `JIT/jit64/hfa/main/testA/hfa_sd0A_r/hfa_sd0A_r.exe`)

Before #13023 merged
```
  N117.                    PUTARG_STK [+0x00]
  consume=2 produce=0
at start of tree, map contains: { N115. COPY -> (116.N115, 116.N115) }
<RefPosition #208 @117 RefTypeUse <Ivl:312> BB01 regmask=[r0-r9 r12 lr] last>
<RefPosition #209 @117 RefTypeUse <Ivl:313> BB01 regmask=[r0-r9 r12 lr] last>

  N119. t3118           =  CNS_INT(h) 0xD1FFAB1E [ICON_STR_HDL] REG NA
  consume=0 produce=1
```

After #13023 merged
```
  N117.                    PUTARG_STK [+0x00]
  consume=2 produce=0
at start of tree, map contains: { N115. COPY -> (116.N115, 116.N115) }
<RefPosition #208 @117 RefTypeFixedReg <Reg:r0 > BB01 regmask=[r0]>
<RefPosition #209 @117 RefTypeUse <Ivl:312> BB01 regmask=[r0] last fixed>
<RefPosition #210 @117 RefTypeFixedReg <Reg:r1 > BB01 regmask=[r1]>
<RefPosition #211 @117 RefTypeUse <Ivl:313> BB01 regmask=[r1] last fixed>

  N119. t3118           =  CNS_INT(h) 0xD1FFAB1E [ICON_STR_HDL] REG NA
  consume=0 produce=1
```
This is the part that putting argument for a call to `public static bool CheckResult(string testName, double actual, double expected)`, and the argument must be third one since it is the only one that goes to the stack.
It seems like the registers shouldn't be fixed since it is PutArgStk(not PutArgReg).

**Comment by wateret** [comment]
Closing since fixed by #13332
